### PR TITLE
Testing enhancements

### PR DESCRIPTION
**Pull request by kiverso**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Migration 
- [ ] Styling
- [x] Testing

### Commit Emojis:
:bar_chart: new tests

### Detailed Description:
Adds happy and sad path testing for all endpoints.
### Why is this change required? What problem does it solve?
This adds tests to get us to the requisite testing threshold and gives us the opportunity to be more confident in our API endpoints.

### Files modified:
- [ main.py] 
- [ app.py] 
- [ tests/* ] 
- [.coveragerc ] 

### Coverage
```Name        Stmts   Miss  Cover
-------------------------------
app.py        129    109    16%
main.py        50     10    80%
models.py      93     13    86%
-------------------------------
TOTAL         272    132    51%```

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[run]
+source = .
+omit =./tests/*,./env/*,manage.py,config.py
+
+[report]
+omit = ./env/*,./tests/*
```
Comment (kiverso) [inline .coveragerc:6]: By default all the packages and files are included in coverage stats, this config file makes it so we only include what we wrote in the coverage stats.

File: main.py [hunk 2]
```diff
@@ -11,15 +11,21 @@ def hello():
 def get_bodies():
     try:
         bodies = CelestialBodies.query.all()
-        return jsonify({'data': [e.serialize() for e in bodies]})
+        if not bodies:
+          return jsonify({'errors': 'No celestial bodies found.'}), 404
+        else:
+          return jsonify({'data': [e.serialize() for e in bodies]})
```
Comment (kiverso) [inline main.py:17]: I added some minor sad paths to make it easier for anyone hitting the API to know what the error is and hopefully communicate what is broken to us more easily.

File: main.py [hunk 3]
```diff
@@ -11,15 +11,21 @@ def hello():
...
 
 @main.route('/api/v1/celestial_bodies/<id>', methods = ['GET'])
 def get_one_body(id):
     try:
         celestial_body = CelestialBodies.query.get(id)
-        return jsonify(celestial_body.serialize())
+        if celestial_body is not None:
+          return jsonify(celestial_body.serialize())
+        else:
+          return jsonify({'errors': f'No object found with id: {id}.'}), 404
```
Comment (kiverso) [inline main.py:28]: Add sad path if there is no celestial body for the requested id.

File: main.py [hunk 4]
```diff
@@ -35,14 +41,20 @@ def get_api():
 def get_landmarks(id):
     try:
         landmarks = Landmark.query.filter(Landmark.celestial_body_id == id)
-        return jsonify({'data': [e.serialize() for e in landmarks]})
+        if CelestialBodies.query.get(id) is not None:
+          return jsonify({'data': [e.serialize() for e in landmarks]})
+        else:
+          return jsonify({'errors': f'No celestial body with id: {id}.'}), 404
```
Comment (kiverso) [inline main.py:47]: Again, I added sad paths for the most common error I could see which is requesting landmarks for a nonexistent celestial body. If no landmarks exist it simply returns an empty list.

File: tests/test_news.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import json
+
+def test_news(app):
+  client = app.test_client()
+  resp = client.get('/api/v1/news')
+  data = data = json.loads(resp.data.decode())
+  assert resp.status_code == 200
+  assert (data['docs'][0]['featured_image'])
+  assert (data['docs'][0]['title'])
```
Comment (kiverso) [inline tests/test_news.py:9]: I just tested that the fields most needed by the FE team are present and that they adhere to the readme for how they will be nested as the news can change each time the test is run.

File: app.py [hunk 6]
```diff
@@ -18,50 +18,6 @@ def create_app():
...
-#         return(str(e))
-
-# @app.route('/api/v1/landmarks/<id>', methods = ['GET'])
-# def get_landmark(id):
-#     try:
-#         landmark = Landmark.query.get(id)
-#         return jsonify(landmark.serialize())
-#     except Exception as e:
-#         return(str(e))
-
```
Comment (arpariseau) [inline app.py:64]: Good to clean up the code.

**Review by arpariseau — APPROVED** [review]

**Pull request merged**